### PR TITLE
add parser for quagga show ip bgp

### PR DESCRIPTION
**Pull request by fabianfrz**
@fichtner I know the script is not beautiful but the output of quagga sadly is that hard to parse (I hate it when programs create tables with empty fields...)

@mimugmail can you run the script and look at the output?
The output should be JSON and from this JSON the GUI should be built.

### CODE REVIEW

**Comment by fabianfrz** [comment]
@fichtner this is the wrapper for @mimugmail for `show ip bgp`

**Comment by mimugmail** [comment]
Hi,

this is the output on my side:

```
root@OPNsense:~ # ./test.rb
{"table_version":"0","local_router_id":"192.168.0.1","output":[{"status":[{"dn":"valid","abb":"*"},{"dn":"best","abb":">"}],"Network":"192.168.0.0","Next Hop":"0.0.0.0","Metric":"0","LocPrf":"","Path":[{"dn":"IGP","abb":"i"}]},{"status":[{"dn":null,"abb":"D"},{"dn":"internal","abb":"i"},{"dn":"suppressed","abb":"s"}],"Network":"played  1 out of","Next Hop":"1 total prefixes","Metric":"","LocPrf":"","Path":[]}]}
```

**Comment by fabianfrz** [comment]
@fichtner please don't merge it to stable. It is @mimugmail to build the diagnostics page -> testing.

**Comment by mimugmail** [comment]
@fabianfrz I had a deeper look at the json results, the last line which informs how many networks were displayed are treated as a network itself.

So when advertising 1 network I have 2 entries, when adding a second one I have 3.

```
root@OPNsense:~ # /usr/local/opnsense/scripts/quagga/diag-bgp.sh
Usage: /usr/local/opnsense/scripts/quagga/diag-bgp.sh bgp|summary|neighbor <ip>|neighbor-adv <ip>
root@OPNsense:~ # /usr/local/opnsense/scripts/quagga/diag-bgp.sh bgp
BGP table version is 0, local router ID is 192.168.0.1
Status codes: s suppressed, d damped, h history, * valid, > best, = multipath,
              i internal, r RIB-failure, S Stale, R Removed
Origin codes: i - IGP, e - EGP, ? - incomplete

   Network          Next Hop            Metric LocPrf Weight Path
*> 192.168.0.0      0.0.0.0                  0         32768 i

Displayed  1 out of 1 total prefixes
root@OPNsense:~ # /usr/local/opnsense/scripts/quagga/diag-bgp.sh bgp
BGP table version is 0, local router ID is 192.168.0.1
Status codes: s suppressed, d damped, h history, * valid, > best, = multipath,
              i internal, r RIB-failure, S Stale, R Removed
Origin codes: i - IGP, e - EGP, ? - incomplete

   Network          Next Hop            Metric LocPrf Weight Path
*> 10.55.55.0/24    0.0.0.0                  0         32768 i
*> 192.168.0.0      0.0.0.0                  0         32768 i

Displayed  2 out of 2 total prefixes
root@OPNsense:~ # /usr/local/opnsense/scripts/quagga/bgp.rb
{"table_version":"0","local_router_id":"192.168.0.1","output":[{"status":[{"dn":"valid","abb":"*"},{"dn":"best","abb":">"}],"Network":"10.55.55.0/24","Next Hop":"0.0.0.0","Metric":"0","LocPrf":"","Path":[{"dn":"IGP","abb":"i"}]},{"status":[{"dn":"valid","abb":"*"},{"dn":"best","abb":">"}],"Network":"192.168.0.0","Next Hop":"0.0.0.0","Metric":"0","LocPrf":"","Path":[{"dn":"IGP","abb":"i"}]},{"status":[{"dn":null,"abb":"D"},{"dn":"internal","abb":"i"},{"dn":"suppressed","abb":"s"}],"Network":"played  2 out of","Next Hop":"2 total prefixes","Metric":"","LocPrf":"","Path":[]}]}
```

**Comment by fabianfrz** [comment]
looks like the parser runs into the line `Displayed  2 out of 2 total prefixes` and interprets it as well. I can fix that if you want.

**Comment by fabianfrz** [comment]
The real thing is that I will integrate it into quagga.rb so it will not be a standalone script anymore.

**Comment by mimugmail** [comment]
Great! But the json output will stay the same for this so I can do further testing?

**Comment by fabianfrz** [comment]
It should except that this line will be gone in the final version and the path in the service file will change. BTW: thanks for reporting
